src/context/WalletProvider/KeepKey/components/Pin.tsx

```diff
@@ -147,6 +182,8 @@ export const KeepKeyPin = () => {
           variant='filled'
           mb={6}
           autoComplete='one-time-code'
+          onKeyDown={handleKeyboardInput}
+          onKeyUp={() => setIsPinEmpty(!pinFieldRef.current?.value)}
```

Comment:
Is there a reason we don't set this on `onKeyDown`, inside `handleKeyboardInput`?

Reply:
Again, I was having out of sync issue here as well with `isPinEmpty`, this seemed to be the best solution.